### PR TITLE
Subtitle support for chromecast enabled. Subtitle handling improved. SRT to VTT conversion added.

### PR DESCRIPTION
**Pull request by mccbala**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

1. Enabled subtitle support for chromecast.
2. Refactored subtitle handling code.
3. Added automatic SRT to VTT conversion support. 

**Which issue (if any) does this pull request address?**
I didn't see any open issue specifying this problem but I saw another pull request addressing part of this problem. This pull request is a slightly improved version and has a bit more functionality.

**Is there anything you'd like reviewers to focus on?**
Please review the handling of subtitles and suggest better practices, if necessary. I have retained `ecstatic` server though it is deprecated by the author. We have to look for an alternative soon.

### CODE REVIEW

**Comment by kalkun** [comment]
[This trailing comma](https://github.com/mccbala/webtorrent-cli/blame/master/package.json#L33) in your package.json prevents it from being installed by npm/yarn

**Comment by mccbala** [comment]
Thanks for the catch. I have removed the trailing comma.

> [This trailing comma](https://github.com/mccbala/webtorrent-cli/blame/master/package.json#L33) in your package.json prevents it from being installed by npm/yarn

**Comment by mccbala** [comment]
> [This trailing comma](https://github.com/mccbala/webtorrent-cli/blame/master/package.json#L33) in your package.json prevents it from being installed by npm/yarn

My bad! Got it.. Perhaps second time's the charm! :)

File: package.json [hunk 1]
```diff
@@ -27,9 +27,10 @@
     "open": "^7.0.0",
     "parse-torrent": "^7.0.0",
     "prettier-bytes": "^1.0.3",
+    "srt-to-vtt": "^1.1.3",
     "vlc-command": "^1.0.0",
     "webtorrent": ">=0.107.6",
-    "winreg": "^1.0.1"
+    "winreg": "^1.0.1"
```
Comment (DiegoRBaquero) [inline package.json:33]:
```suggestion
    "winreg": "^1.0.1"
```

**Comment by DiegoRBaquero** [comment]
Would be great to get this going, can you rebase and re-test?

**Pull request closed without merging**